src/gui/qgsexpressionbuilderwidget.h

```diff
@@ -380,6 +380,55 @@ class GUI_EXPORT QgsExpressionBuilderWidget : public QWidget, private Ui::QgsExp
      */
     void removeSelectedUserExpression( );
 
+    /**
+     * Edits the selected expression from the stored user expressions,
+     * the selected expression must be a user stored expression.
+     * \since QGIS 3.14
+     */
+    void editSelectedUserExpression();
+
+    /**
+     * Display a file dialog to choose where to store the exported expressions JSON file
+     * and saves them to the selected destination.
+     * \since QGIS 3.14
+     */
+    void exportUserExpressions_pressed();
+
+    /**
+     * Create the expressions JSON document storing all the user expressions to be exported.
+     * \returns the created expressions JSON file
+     * \since QGIS 3.14
+     */
+    QJsonDocument *exportUserExpressions();
```

Comment:
If this transfers ownership, it should return a std::unique_ptr

Reply:
Can you please review if I made proper use of `str::unique_ptr`.